lib/order.rb

```diff
@@ -0,0 +1,89 @@
+require_relative "customer"
+
+class Order
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products #hash
+    @customer = customer #instance of Customer
+    valid_statuses = [:pending, :paid, :processing, :shipped, :complete]
+    if !valid_statuses.include? fulfillment_status
+      raise ArgumentError.new("Invalid status")
+    end
```

Comment:
This might be clearer if it was formatted like a guard clause.  Also, it's helpful for debugging to include the bad argument in the error message.

```suggestion
    raise ArgumentError.new("Invalid status: #{status}") unless valid_statuses.include? fulfillment_status
```